src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/Et3VettingHelper.java

```diff
@@ -294,4 +314,67 @@ private static boolean respondentExtensionExists(RespondentSumType respondent) {
         return YES.equals(respondent.getExtensionRequested()) && YES.equals(respondent.getExtensionGranted());
     }
 
+    /**
+     * Finds listed hearings for a case and sets the hearing details for ExUI. Will display a table with the earliest
+     * hearing date and track type or static text saying that there are no listings for the case.
+     * @param caseData data for the current case
+     */
+    public static void checkHearingListed(CaseData caseData) {
+        if (CollectionUtils.isEmpty(caseData.getHearingCollection())) {
+            log.info(String.format("No hearings for case %s", caseData.getEthosCaseReference()));
+            caseData.setEt3HearingDetails(CASE_NOT_LISTED);
+            caseData.setEt3IsCaseListedForHearing(NO);
+            return;
+        }
+
+        String hearingDate = findHearingDate(caseData.getHearingCollection());
+
+        if (CASE_NOT_LISTED.equals(hearingDate)) {
+            caseData.setEt3HearingDetails(CASE_NOT_LISTED);
+            caseData.setEt3IsCaseListedForHearing(NO);
+            return;
+        }
+
+        String track = isNullOrEmpty(caseData.getConciliationTrack())
+            ? "Track could not be found"
+            : caseData.getConciliationTrack();
+
+        caseData.setEt3HearingDetails(String.format(ET3_HEARING_TABLE, hearingDate, track));
+        caseData.setEt3IsCaseListedForHearing(YES);
+    }
+
+    private static String findHearingDate(List<HearingTypeItem> hearingCollection) {
+        List<String> hearingDates = new ArrayList<>();
+
+        for (HearingTypeItem hearingTypeItem : hearingCollection) {
+            for (DateListedTypeItem dateListedTypeItem : hearingTypeItem.getValue().getHearingDateCollection()) {
+                if (HEARING_STATUS_LISTED.equals(dateListedTypeItem.getValue().getHearingStatus())) {
+                    hearingDates.add(dateListedTypeItem.getValue().getListedDate());
+                }
+            }
+        }
```

Comment:
Instead of this, try something like this:

```
hearingCollection.stream().foreach(item -> addListedDates(hearingDates, item.getValue().getHearingDateCollection());


private void addListedDates(List<String> hearingDates, Collection hearingCollection) {
    hearingCollection
      .stream()
      .filter(item -> HEARING_STATUS_LISTED.equals(item.getValue().getHearingStatus())
      .forEach(hearingDates.add(item.getValue().getListedDate()));
}
```